pyproject.toml

```diff
@@ -1,3 +1,8 @@
 [build-system]
-requires = ["setuptools", "wheel", "oldest-supported-numpy", "cython>=0.23"]
-build-backend = "setuptools.build_meta"
\ No newline at end of file
+requires = [
+    "setuptools>=42",
+    "oldest-supported-numpy; python_version < '3.9'",
+    "numpy>=2.0.0; python_version >= '3.9'",
```

Comment:
@henryiii mentioned that a `#define` is needed here for this to work as well, but I'll need to check how NumPy uses this.

Reply:
Though it seems this is only if NumPy C-API headers are exposed: https://numpy.org/doc/stable/dev/depending_on_numpy.html#build-time-dependency So maybe this is okay as is.

Reply:
> Thus, we guarantee that, when using defaults, NumPy 1.25 will expose a C-API compatible with NumPy 1.19

Ah, looks like the default is actually reasonable. I'd forgotten that it was not "latest" like it was before this change. (Pybind11 people like me don't have to worry about this at all :P )